opm/output/eclipse/AggregateGroupData.hpp

```diff
@@ -0,0 +1,162 @@
+/*
+  Copyright (c) 2018 Statoil ASA
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_AGGREGATE_GROUP_DATA_HPP
+#define OPM_AGGREGATE_GROUP_DATA_HPP
+
+#include <opm/output/eclipse/CharArrayNullTerm.hpp>
+#include <opm/output/eclipse/WindowedArray.hpp>
+
+#include <cstddef>
+#include <string>
+#include <vector>
+#include <map>
+
+namespace Opm {
+    class Schedule;
+    class SummaryState;
+    class Group;
+} // Opm
+
+namespace Opm { namespace RestartIO { namespace Helpers {
+
+    class groupMaps {
+    public:
+	const std::map <size_t, const Opm::Group*>& indexGroupMap() const;
+	const std::map <const std::string, size_t>& groupNameIndexMap() const;
+
+	void currentGrpTreeNameSeqIndMap(const Opm::Schedule&                        sched,
+	                                 const size_t                                simStep,
+	                                 const std::map<const std::string , size_t>& GnIMap,
+	                                 const std::map<size_t, const Opm::Group*>&  IGMap);
+
+    private:
+	std::map <size_t, const Opm::Group*> m_indexGroupMap;
+	std::map <const std::string, size_t> m_groupNameIndexMap;
+    };
+
+    class AggregateGroupData
+    {
+    public:
+	explicit AggregateGroupData(const std::vector<int>& inteHead);
+
+	void captureDeclaredGroupData(const Opm::Schedule&                 sched,
+				      const std::vector<std::string>&      restart_group_keys,
+				      const std::vector<std::string>&      restart_field_keys,
+				      const std::map<std::string, size_t>& groupKeyToIndex,
+				      const std::map<std::string, size_t>& fieldKeyToIndex,
+				      const std::size_t                    simStep,
+				      const Opm::SummaryState&             sumState,
+				      const std::vector<int>&              inteHead);
+
+        /// Retrieve Integer Group Data Array.
```

Comment:
Please stop these completely uninformative comments - it is just noise.

Reply:
Doxygen.

Reply:
????

Reply:
Doxygen requires these comments to create reference documentation in HTML.

Reply:
> Doxygen requires these comments to create reference documentation in HTML.

yes - of course I understand that, but I can not see that has been agreement on facilitating doxygen output. Iff we indeed agree on that at some stage I will of course support that; but right now I can not see that is the case.

I think stylistic issues is not that important, and we should all get along and so on; but I do get a little frustrated that you seem to consistently ignore my review comments on style issues. Style issues are subjective, and to facilitate good cooperation there must be some give and take on the matter.

Reply:
> but I do get a little frustrated that you seem to consistently ignore my review comments on style issues.

I don't ignore them&mdash;I'm busy addressing them in the context of #451 (especially literal vs. named vector elements) and expect to publish a set of updates in a few hours&mdash;but it's clear we disagree on braces for single-statement bodies of control structures and in the absence of convincing evidence I don't think I'll change my view on that subject.  I certainly appreciate the economy of vertical space you get from not having having braces, but I think the drawbacks outweigh those benefits.

For one, it's easier to make a mistake in updating conditionals.  Consider the not entirely contrived example of hastily changing
```C++
if (some_condition)
    do_A();
else
    do_B();
```
into
```C++
if (some_condition)
    if (other_condition)
        do_A();
else
    do_B();
```
In this case the second version will `do_B()` when `some_condition && !other_condition` whereas the first would `do_B()` when `!some_condition`.  Now that may certainly be the correct conditions, but the indentation does at least **suggest** otherwise.  Of course recent versions of GCC will warn about misleading indentation (`-Wmisleading-indentation`, part of `-Wall` since GCC 6) but then you have to pay close attention to the warnings and if you're making quick changes you might not be.  Not a problem if you have perfect test coverage of the relevant code section.

Having the braces **might** also reduce the risk of something like #359 being introduced, although that's speculation on my part.  Finally, tooling like KDevelop (or Emacs for that matter) have an easier time collapsing/folding code sections that are delimited by braces than bare statements.  That's certainly tailoring your code to your tools, but might not be entirely inconsequential nonetheless.  I know that I appreciate that ability when I'm getting into new code bases and am trying to navigate the high-level logic and structure.

Reply:
I have removed the /// comment(s) mentioned by Joakim. Else I leave the discussion on the code adaption to facilitate Doxygen output to the maintainers and will follow the recommended practice when agreed.

Reply:
> I have removed the /// comment(s) mentioned by Joakim.

Thank you.

Reply:
> I don't ignore them—I'm busy addressing them in the context of #451 (especially literal vs. named vector elements)

OK - good. 

> but it's clear we disagree on braces for single-statement bodies of control structures and in the absence of convincing evidence I don't think I'll change my view on that subject.

Yes - that is clearly true. I understand your arguments - and I agree they have merit; they just don't weigh up enough in my book. I have also come to realize that my opinion on matters like this sometimes change, I have come to prefer snake_case to CamelCase (I realize that ship has left for OPM), short(er) method names and recently I have even experimented with public data members. I consider being able to change your mind a virtue - maybe one day I will add braces to one line statements.

But my main point is that rational arguments aside there will *always* be a level subjectiveness to these questions, and to get a good review process we must sometimes leave the fully objective horse in the stable and give and take a bit when it comes to style issues.